smart-contracts/contracts/cl-vault/src/vault/swap.rs

```diff
@@ -64,8 +75,12 @@ pub fn swap(
     token_in_amount: Uint128,
     token_in_denom: &String,
     token_out_min_amount: Uint128,
+    token_out_denom: &String,
+    recommended_swap_route: Option<SwapOperationsListUnchecked>,
```

Comment:
Do we want to nest the recommended_swap_route etc in the swap function here?

Reply:
Can you explain